### PR TITLE
Change activity to context in Lock initialization

### PR DESCRIPTION
**Pull request by skrabacz-michal**

### CODE REVIEW

**Comment by skrabacz-michal** [comment]
@lbalmaceda please review

**Comment by lbalmaceda** [comment]
@skrabacz-michal why do you need to change this?

**Comment by skrabacz-michal** [comment]
@lbalmaceda there is no need to bind Lock with Activity (at least I didn't find any) - using Context is more flexible (e.g for testing). Right now I try to create simple wrapper around Lock with Rx support and providing Activity is pain in the ass ;)

**Comment by lbalmaceda** [comment]
I did a quick check on my side and didn't find any need for an `Activity` context today. Let's move it to `Context`, a few things to consider:
- `Lock` as well as `PasswordlessLock` have similar initialization methods. You'll need to update them all:
   - Builder#Build(Activity)
   - #newIntent(Activity)
   - #onDestroy(Activity)
- Ensure that javadoc now talks about a `Context` instead of an `Activity`.

**Comment by skrabacz-michal** [comment]
@lbalmaceda done

File: lib/src/main/java/com/auth0/android/lock/Lock.java [hunk 1]
```diff
@@ -196,14 +196,14 @@ public Builder(Auth0 account, LockCallback callback) {
          * Finishes the construction of the Lock.Options and generates a new Lock instance
          * with those Lock.Options.
          *
-         * @param activity a valid Activity context
+         * @param context a valid Activity context
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/Lock.java]: "a valid Context"

File: lib/src/main/java/com/auth0/android/lock/Lock.java [hunk 2]
```diff
@@ -114,34 +113,35 @@ public static Builder newBuilder(@NonNull LockCallback callback) {
...
-        Intent lockIntent = new Intent(activity, LockActivity.class);
+    public Intent newIntent(Context context) {
+        Intent lockIntent = new Intent(context, LockActivity.class);
         lockIntent.putExtra(Constants.OPTIONS_EXTRA, options);
+        lockIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         return lockIntent;
     }
 
     /**
      * Should be called on the Activity holding the Lock instance's OnDestroy method, as it
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/Lock.java]: "This method ensures proper Lock's lifecycle handling. Must be called from the class holding the Lock instance whenever you're done using it. i.e. in the Activity's onDestroy method."

File: lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java [hunk 3]
```diff
@@ -110,33 +109,34 @@ public static Builder newBuilder(@NonNull LockCallback callback) {
...
-        Intent lockIntent = new Intent(activity, PasswordlessLockActivity.class);
+    public Intent newIntent(Context context) {
+        Intent lockIntent = new Intent(context, PasswordlessLockActivity.class);
         lockIntent.putExtra(Constants.OPTIONS_EXTRA, options);
+        lockIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         return lockIntent;
     }
 
     /**
      * Should be called on the Activity holding the Lock instance's OnDestroy method, as it
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java]: "This method ensures proper Lock's lifecycle handling. Must be called from the class holding the Lock instance whenever you're done using it. i.e. in the Activity's onDestroy method."

File: lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java [hunk 4]
```diff
@@ -110,33 +109,34 @@ public static Builder newBuilder(@NonNull LockCallback callback) {
...
      * ensures the correct Lock lifecycle handling.
      *
-     * @param activity a valid Activity context
+     * @param context a valid Activity context
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java]: "a valid Context"

File: lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java [hunk 5]
```diff
@@ -187,14 +187,14 @@ public Builder(Auth0 account, LockCallback callback) {
          * Finishes the construction of the Lock.Options and generates a new Lock instance
          * with those Lock.Options.
          *
-         * @param activity a valid Activity context
+         * @param context a valid Activity context
```
Comment (lbalmaceda) [inline lib/src/main/java/com/auth0/android/lock/PasswordlessLock.java]: "a valid Context"

**Comment by skrabacz-michal** [comment]
@lbalmaceda could u check?

**Comment by lbalmaceda** [comment]
@skrabacz-michal done. Thanks.

**Comment by skrabacz-michal** [comment]
@lbalmaceda could u merge it? 👍

**Pull request merged**